packages/vector_graphics_compiler/lib/src/svg/node.dart

```diff
@@ -452,13 +466,109 @@ class TextNode extends AttributedNode {
   }
 
   @override
-  void build(DrawCommandBuilder builder, AffineMatrix transform,
-      Rect nearestParentBounds) {
-    final Paint? paint = _paint(nearestParentBounds, transform);
-    final TextConfig textConfig = _textConfig(nearestParentBounds, transform);
-    if (paint != null && textConfig.text.trim().isNotEmpty) {
-      builder.addText(textConfig, paint, attributes.id);
-    }
+  void visitChildren(NodeCallback visitor) {}
+
+  @override
+  S accept<S>(Visitor<S> visitor) {
+    return visitor.visitTextNode(this);
+  }
+}
+
+/// A block of text that has its position and final transfrom fully known.
+class ResolvedTextNode extends Node {
+  /// Create a new [ResolvedTextNode].
+  ResolvedTextNode({
+    required this.textConfig,
+    required this.paint,
+  });
+
+  /// The text configuration to draw this piece of text.
+  final TextConfig textConfig;
+
+  /// The paint used to draw this piece of text.
+  final Paint paint;
+
+  @override
+  S accept<S>(Visitor<S> visitor) {
+    return visitor.visitResolvedText(this);
+  }
+
+  @override
+  void visitChildren(NodeCallback visitor) {}
+}
+
+/// A path node that has its bounds fully computed.
+class ResolvedPathNode extends Node {
```

Comment:
Would it be better to define these in the file with the visitor that creates them? 

What I'd really like here is something that future us working on the parser will remember to never instantiate these from the parser. One way to achieve that is to define them in the same file as the resolver and make the ctor's private.

Reply:
I think we can add a comment no? But I can move them and the resolver into a new library

Reply:
Moved to resolver.dart and added comments